### ISSUE TITLE
"List index (3) out of bounds" when backspacing new title over default title

### ISSUE THREAD

**Issue by mlncn** [tomboy-notes/tomboy-ng#30]
1. Create a new note
2. Paste or type text into the body of the note.
3. Backspace from the start of this text over the title to replace the default title with this text in the note
4. Received this error:

> List index (3) out of bounds.
> 
> Press OK to ignore and risk data corruption.
> Press Abort to kill the program.

5. Pressed OK.
6. Note is unresponsive.  Cannot click or type in title or body.  Oh.  Actually, all of Tomboy is unresponsive/broken.  It is not using any system resources so not much hope of it coming back either.

Nope.  Never came back.  Guess i should have clicked "Abort"!

I have the notes in Git and no data corruption is evident.

Environment/context: Same as #29 ... sorry.

**Comment by mlncn** [comment]
The program stayed frozen; trying to close any note prompted a force quit message which quit the whole program.

Here's the newly created note that it bailed on, after i pasted in the "Improve performance.." line and started backspacing over the default note title.:

```
<?xml version="1.0" encoding="utf-8"?>
<note version="0.3" xmlns:link="http://beatniksoftware.com/tomboy/link" xmlns:size="http://beatniksoftware.com/tomboy/size" xmlns="http://beatniksoftware.com/tomboy">
  <title>New Note 2018-02-04 14:36:49.666Improve performance for large note collections</title>
  <text xml:space="preserve"><note-content version="0.1">New Note 2018-02-04 14:36:49.666Improve performance for large note collections
</note-content></text>
  <last-change-date>2018-02-04T14:37:17.3310000-05:00</last-change-date>
  <last-metadata-change-date>2018-02-04T14:37:17.3310000-05:00</last-metadata-change-date>
  <create-date>2018-02-04T14:37:17.3310000-05:00</create-date>
  <cursor-position>1</cursor-position>
  <selection-bound-position>1</selection-bound-position>
  <width>1000</width>
  <height>626</height>
  <x>0</x>
  <y>0</y>
  <open-on-startup>False</open-on-startup>
</note>
```

**Comment by davidbannon** [comment]
Thanks for your reports Ben, especially this one !
I 'thought' I saw exactly this while I was going through my test process for recent release but could not reproduce it. At the time I had backspaced across all of the title and then auto save cut in. Tried several times to do it again but could not. Since then, I have locked KMemo during a save and (obviously) have not seen it since. Not very conclusive.
So, that fix (if its a fix) is in the Git code but not in the binaries yet. I'm going to do a new binary for the Mac, maybe I make it a new binaries all round ?  A day or so ...
